Assembly-CSharp/Global/Sound/VoicePlayer.cs

```diff
@@ -25,6 +26,82 @@ public VoicePlayer()
 		this.stateTransition.Add(new SdLibSoundProfileStateGraph.TransitionDelegate(base.StopSound), SoundProfileState.CrossfadeIn, SoundProfileState.Stopped);
 	}
 
+	public static void PlayBattleVoice(int va_id, string text)
+    {
+		string vaPath = String.Format("Voices/{0}/battle/{2}/va_{1}", Localization.GetSymbol(), va_id, FF9StateSystem.Battle.battleMapIndex);
+		if (!String.IsNullOrEmpty(AssetManager.SearchAssetOnDisc("Sounds/" + vaPath + ".akb", true, true)) || !String.IsNullOrEmpty(AssetManager.SearchAssetOnDisc("Sounds/" + vaPath + ".ogg", true, false)))
+		{
+			SoundLib.VALog(String.Format("field:battle/{0}, feild:{0}, msg:{1}, text:{2} path:{3}", FF9StateSystem.Battle.battleMapIndex, va_id, text, vaPath));
```

Comment:
On this line, we are mixing abstractions levels. On the one hand, you are posting a log to `SoundLib` (higher order abstraction) but on the other, you are manipulating strings (lower order abstraction). Move your message creation logic to separate method the keep same level of abstraction.

```suggestion
			int battleMapIndex = FF9StateSystem.Battle.battleMapIndex;
			string msg = CreateLogMsg(battleMapIndex, va_id, text, vaPath);
			SoundLib.VALog(msg);
```
```csharp
private string CreateLogMsg(int battleMapIndex, int voiceActorId, string text, string path) {
    if (battleMapIndex == -1) {
        return $"field:battle/general, msg:{voiceActorId}, text:{text} path:{path}";
    } else {
        return $"field:battle/{battleMapIndex}, field:{battleMapIndex}, msg:{voiceActorId}, text:{text} path:{path}"
    }
}
```

Note, you can do the same for the `else` clause as well.